pkg/operator/sync.go

```diff
@@ -94,30 +94,26 @@ func (c *csiDriverOperator) syncDeployment(instance *v1alpha1.Driver) (*appsv1.D
 func (c *csiDriverOperator) syncDaemonSet(instance *v1alpha1.Driver) (*appsv1.DaemonSet, error) {
 	daemonSet := c.getExpectedDaemonSet(instance)
 
-	// Update the daemonSet when something updated Driver.Spec.LogLevel.
-	// The easiest check is for Generation update (i.e. redeploy on any Driver.Spec change).
-	// This may update the DaemonSet more than it is strictly necessary, but the overhead is not that big.
-	forceRollout := false
-	if instance.Generation != instance.Status.ObservedGeneration {
-		forceRollout = true
-	}
-
-	if c.versionChanged("operator", c.operatorVersion) {
-		// Operator version changed. The new one _may_ have updated DaemonSet -> we should deploy it.
-		forceRollout = true
+	// Record the hash of the spec in an annotation so ApplyDaemonSet
+	// below detects its change. This is going to cover log level, image
+	// and/or bindata changes.
+	// TODO: use resourceapply.ApplyDaemonSet when it gets hashing of Spec.
+	jsonBytes, err := json.Marshal(daemonSet.Spec)
+	if err != nil {
+		return nil, err
 	}
-
-	if c.versionChanged("aws-ebs-csi-driver", c.operandVersion) {
-		// Operand version changed. Update the deployment with a new image.
-		forceRollout = true
+	specHash := fmt.Sprintf("%x", sha256.Sum256(jsonBytes))
+	if daemonSet.Annotations == nil {
+		daemonSet.Annotations = map[string]string{}
 	}
+	daemonSet.Annotations[specHashAnnotation] = specHash
```

Comment:
I'd suggest moving this to the `getDaemonSet()` function down below (`getDeployment() for the deployment). So far we've been "patching" the objects only in those functions.

Reply:
I expect the hashing to be done by ApplyDaemonSet soon, so I want to keep it close.